File: 4. Funciones/funcionmain.cpp
#include <iostream> //se añaden los archivos que se quieren incluir dentro de nuestra funcion
int main ()  //se define una funcion declarando su tipo, en este caso se define una funcion de enteros
{   //usamos corchetes para encapsular toda nuestra funcion en estos
    return 0; //retornamos un valor 0 para decir que es el fin de nuestra funcion
} //cerramos los corchetes que marcan el final de la funcion


    
/*
En c++ las funciones funcionan en base a corchetes, esto quiere decir que todo lo que haya dentro de estos son lo que representa la funcion como tal,
bajo esta premisa entonces podemos decir lo siguiente: 
    -No importan las identaciones (tabs)
    -Si quisieramos podriamos escribir todo dentro de la misma linea
Entonces bajo esta premisa podemos denotar que el siguiente ejemplo es lo mismo que la funcion de arriba
*/

int funcion(){printf("Holamundo"); return 0;}

//Asi como tambien podemos decir que lo siguiente es igual que lo de arriba

int funcion2(){
printf("Hola mundo")
    return 0;
}

/*
Ya, pero por que pasa esto?
Esto pasa porque el compilador interpreta las lineas de codigo en base a los operadores logicos que hay, de esta forma es que cuando pasamos los corchetes 
nos referimos a todo lo que hay dentro de una funcion, por esto mismo podemos decir que el compilador entiende como linea de codigo todo lo que hay hasta llegar
a un ";" (punto y coma), entonces todo lo que haya separado bajo este operador, cuenta dentro de la misma linea de codigo

*/

File: 4. Funciones/funcionmain.py
# Definicion de una funcion en python
# para definir una funcion usamos el identificador "def" seguido de "nombreDeLaFuncion()" y un ":"
def main():
    return 0
#al terminar la funcion retornamos un 0 al igual que en c++

# IMPORTANTE
# def main():
#     esto esta dentro de la funcion
# esto esta fuera de la funcion y da un error
#     return 0
#
# Python no funciona con llaves, funciona con identaciones
# Esto quiere decir que cuando definas una funcion uses el "enter" y te asegures de usar el "tab" para que todo quede dentro de una funcion
# ejemplo de esto es lo siguiente

def funcion():
    print("Estoy dentro de la funcion ")
print("Estoy fuera de la funcion")

# De esta misma forma se entiende por ejemplo que los if y los ciclos funcionan bajo la misma regla, veamos el siguiente ejemplo de esto

def ciclos():
    print("Estoy dentro de la funcion")
    i = 0
    while(i < 100):
        print("Estoy dentro de la funcion y dentro del ciclo")
        i = i+1
        if (i/2 > 50):
            print("Estoy dentro de la funcion, dentro del ciclo y tambien dentro del if")
        else:
            print("Estoy dentro de la funcion, dentro del ciclo y dentro del condicional else")
        
        print("Estoy dentro de la funcion, dentro del ciclo pero fuera de los condicionales")
    
    print("Estoy dentro de la funcion pero fuera del ciclo")
    
print("Estoy fuera de la funcion, fuera del ciclo y fuera de los condicionales")
